File: password-validator-python/test_password_validator.py
import unittest
from password_validator import validate_password

class TestPasswordValidator(unittest.TestCase):
    def test_fxmedia12345678(self):
        self.assertTrue(validate_password('Fxmedia12345678@'))

    def test_another_secure_pwd(self):
        self.assertTrue(validate_password('AnotherSecurePwd$567'))

    def test_super_passphrase(self):
        self.assertTrue(validate_password('SuperP@ssphrase123'))

    def test_short_pwd(self):
        self.assertFalse(validate_password('ShortPwd1!'))

    def test_password(self):
        self.assertFalse(validate_password('Password'))

    def test_lowercase_only(self):
        self.assertFalse(validate_password('lowercaseonly'))

    def test_missing_upper_and_special(self):
        self.assertFalse(validate_password('MISSINGUPPERANDSPECIAL'))

if __name__ == '__main__':
    unittest.main()

File: password-validator-node/test/testPasswordValidator.js
const { expect } = require('chai');
const { validatePassword } = require('../passwordValidator');

describe('Password Validator', () => {
  it('should validate Fxmedia12345678@ as true', () => {
    expect(validatePassword('Fxmedia12345678@')).to.be.true;
  });

  it('should validate AnotherSecurePwd$567 as true', () => {
    expect(validatePassword('AnotherSecurePwd$567')).to.be.true;
  });

  it('should validate SuperP@ssphrase123 as true', () => {
    expect(validatePassword('SuperP@ssphrase123')).to.be.true;
  });

  it('should validate ShortPwd1! as false (too short)', () => {
    expect(validatePassword('ShortPwd1!')).to.be.false;
  });

  it('should validate Password as false (too short)', () => {
    expect(validatePassword('Password')).to.be.false;
  });

  it('should validate lowercaseonly as false (only one category)', () => {
    expect(validatePassword('lowercaseonly')).to.be.false;
  });

  it('should validate MISSINGUPPERANDSPECIAL as false (too short)', () => {
    expect(validatePassword('MISSINGUPPERANDSPECIAL')).to.be.false;
  });
});